app/Console/Commands/UpdatePostsWithActionIds.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Rogue\Console\Commands;
+
+use DB;
+use Rogue\Models\Post;
+use Rogue\Models\Action;
+use Illuminate\Console\Command;
+
+class UpdatePostsWithActionIds extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'rogue:updatepostswithactionids';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Adds action_ids to all posts pre post metadata.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $this->line('rogue:updatepostswithactionids: Staring script to add action_ids to posts!');
+
+        // Grab all of the actions in the actions table.
+        $actions = Action::all();
+
+        // For each action, find all of its posts and add the action_id to each of them.
+        foreach ($actions as $action) {
+            DB::table('posts')
+                ->where('action', $action->name)
```

Comment:
Won't this be `default` for most `posts`? I thought on the import these would have something more like an internal title, so I wouldn't have thought we'd be finding matches here.

Reply:
@katiecrane yes, most are `default` but there are others too! Some have something like `action-NA`, `action-1`, `affirmation`, `june-2018-turbovote`, etc.

Reply:
@chloealee won't only the ones with an exact match on the action name end up with an `action_id` though?

Reply:
@katiecrane yup! Which we will import from the csv that we pulled that got distinct actions from the posts in the database. I don't think I'm understanding what you're saying here?

Reply:
@chloealee ah sorry for being confusing! I thought we wanted to give every post an `action_id` with this script, but it seems to that posts with `action` of `default` will NOT get an `action_id` here?

Reply:
@katiecrane don't apologize, this is a little confusing! 

I _think_ I understand what you're asking but sorry if this is just repeating what you already know! So we have pulled a CSV that looks like this (dummy data):
![screen shot 2019-01-18 at 5 43 08 pm](https://user-images.githubusercontent.com/9019452/51416686-8e451c80-1b48-11e9-9d41-9734a844aa6d.png)

We first import all of these into the `actions` table so there will be may actions with the name `default` but if they are under different campaigns and/or have different post types, then it will be a different action and have a different `action_id`. 

Then, we run this part of the script and go through each action and find all the posts that the action has (determined if they have the same `campaign_id`, `name`/`action`, and post `type`. Then we'll add that `action_id` to those posts. 

Does that make sense?

Reply:
@chloealee YES OMG THANK YOU!!! I didn't look at your test csv carefully enough, I didn't realize action titles would be `default` when we import!

Reply:
ok yay glad it makes sense!! It could have very well been that I did something wrong too bc I started writing this script way differently initially and I feel like I've gotten mixed up during this! Thanks for confirming logic makes sense! :)

Reply:
Thanks for walking me through it @chloealee !